x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/optimizer/OptimizerRules.java

```diff
@@ -1137,6 +1152,209 @@ protected Expression rule(Expression e) {
         }
     }
 
+    // Simplifies arithmetic expressions with fixed point fields in BinaryComparisons.
+    public static final class SimplifyComparisonsArithmetics extends OptimizerExpressionRule {
+        BiFunction<DataType, DataType, Boolean> typesCompatible;
+
+        public SimplifyComparisonsArithmetics(BiFunction<DataType, DataType, Boolean> typesCompatible) {
```

Comment:
`typesCompatible` -> `typeCheck`

Reply:
Before I change it, just to 2x check: the passed arguments are `DataTypes::areCompatible` and `SqlDataTypes::areCompatible`.
The code using the var also looks like `return typesCompatible.apply([type1], [type2]) == false;`.
Not sure if the change would bring more clarity ("what's checked?"), so asking.